### PR TITLE
use JuliaFormatter

### PR DESCRIPTION
**Pull request by jw3126**
@briochemc @gustaphe 
Here is the promised format PR. Personally, I am not married to a particular style and prefer to leaf everything default over tweaking it. Should I tweak the format, add a formatting script or even enforce the format by a `@test JuliaFormatter.format(package_root)`? Any thoughts?

### CODE REVIEW

**Comment by gustaphe** [comment]
Nice.

Having a `@test` call change code makes me a bit uneasy. But `@test JuliaFormatter.format(package_root, overwrite=false)` will simply fail the test if the code isn't formatted.

**Comment by jw3126** [comment]
Dang the format test fails on some Windows builds. I think the most pragmatic thing is then to just drop it.

File: docs/lit/examples/1_Examples.jl [hunk 1]
```diff
@@ -36,21 +36,21 @@ plot!(y2)
 #
 # But you can add inset subplots with different axes that have different dimensions
 
-plot!(rand(10)*u"m", inset=bbox(0.5, 0.5, 0.3, 0.3), subplot=2)
+plot!(rand(10) * u"m", inset = bbox(0.5, 0.5, 0.3, 0.3), subplot = 2)
```
Comment (briochemc) [inline docs/lit/examples/1_Examples.jl]: I'm not sure these spaces for kwargs look good. What do you think?

File: docs/lit/examples/1_Examples.jl [hunk 2]
```diff
@@ -59,80 +59,85 @@ plot(y, ylabel="")
 
 # `unitformat` can be a boolean or `nothing`:
 
-plot([plot(y, ylab="mass", title=repr(s), unitformat=s) for s in (nothing, true, false)]...)
+plot(
```
Comment (briochemc) [inline docs/lit/examples/1_Examples.jl]: This seems a bit of an overkill, no? IMHO the one-liner looked better before...

File: docs/lit/examples/1_Examples.jl [hunk 3]
```diff
@@ -146,7 +151,7 @@ contourf(x, y, z)
 # which will be converted to the units of `y` and plot your errorbars:
 
 using Unitful: GeV, MeV, c
-x = (1.0:0.1:10) * GeV/c
-y = @. (2 + sin(x / (GeV/c))) * 0.4GeV/c^2 # a sine to make it pretty
-yerror = 10.9MeV/c^2 * exp.(randn(length(x))) # some noise for pretty again
-plot(x, y; yerror, title="My unitful data with yerror bars", lab="")
+x = (1.0:0.1:10) * GeV / c
```
Comment (briochemc) [inline docs/lit/examples/1_Examples.jl:148]: I think I prefered when the units were compact as before. Opinions?

File: docs/lit/examples/2_Plots.jl [hunk 4]
```diff
@@ -18,44 +18,83 @@ using Plots, Unitful, UnitfulRecipes
 
 # ## Lines
 
-plot(Plots.fakedata(50, 5) * u"m", w=3)
+plot(Plots.fakedata(50, 5) * u"m", w = 3)
 
 # ## Parametric plots
 
-plot(t -> sin(t)*u"s", t -> sin(2t)*u"m", 0, 2π, line=4, leg=false, fill=(0, :orange))
+plot(
```
Comment (briochemc) [inline docs/lit/examples/2_Plots.jl]: Same as above, this turns a one-liner into an 9-lines example... IMHO it was better before 🤷‍♂️

File: docs/lit/examples/2_Plots.jl [hunk 5]
```diff
@@ -18,44 +18,83 @@ using Plots, Unitful, UnitfulRecipes
...
+    fill = (0, :orange),
+)
 
 # ## Colors
 
-y = rand(100)*u"km"
-plot((0:10:100)*u"hr", rand(11, 4)*u"km", lab="lines", w=3, palette=:grays, fill=0, α=0.6)
-scatter!(y, zcolor=abs.(y .- 0.5u"km"), m=(:heat, 0.8, Plots.stroke(1, :green)), ms=10 * abs.(y .- 0.5u"km") .+ 4u"km", lab="grad")
+y = rand(100) * u"km"
+plot(
```
Comment (briochemc) [inline docs/lit/examples/2_Plots.jl:30]: Same as above, these examples were nice and compact and look a bit worse to me now.

**Review by briochemc — COMMENTED** [review]
Overall I think this does more harm than good... Not sure what you guys think.
What I see essentially is that examples look more complicated now (and don't match the copy-pasted examples from Plots.jl):
| before | after |
| :---: | :---: |
| Screen Shot 2021-03-18 at 12 05 14 am | Screen Shot 2021-03-18 at 12 04 55 am |

Also I'm not in favor of adding spaces between mulitplications/divisions of units, e.g., replacing `10MeV/c` with `10MeV / c` for example.

Maybe there are options in JuliaFormatter.jl to deal with these things in a nicer way?

(Sorry I don't mean to be the party-pooper, I'm just trying to be honest with my comments but know that I truly appreciate the efforts and I apologize if any — or all — of my comments were annoying!)

**Comment by gustaphe** [comment]
(I'll just continue to give my 2c, hope that's okay)

If the build fails I agree this is not worth it, but for the eventuality that you want to at least do this manually once in a while:

Most of @briochemc 's comments are solved by `format(pkgdir(UnitfulRecipes), BlueStyle())` (specifically, BlueStyle agrees with us both that kwargs should not have spaces, and it's a bit more friendly to one-liners).

One could also consider formatting only the `src` directory, as the `docs` code is more "for show", and likely to be manually optimized for visual effect already.

**Comment by briochemc** [comment]
These are very good points @gustaphe. @jw3126 would you mind trying this with the `BlueStyle()` flag?

Also additional minor point: if we are not testing code formatting in the tests, we should drop the extra dependency, right?

File: docs/lit/examples/2_Plots.jl [hunk 6]
```diff
@@ -72,94 +97,132 @@ histogram2d(randn(10000) * u"cm", randn(10000) * u"cm", nbins=20)
...
+    z;
+    zcolor=reverse(z),
+    m=(10, 0.8, :blues, Plots.stroke(0)),
+    leg=false,
+    cbar=true,
+    w=5,
+    xlabel="x",
+    ylabel="y",
+    zlabel="z",
+)
```
Comment (jw3126) [inline docs/lit/examples/2_Plots.jl:171]: BlueStyle still expands these one-liners.

**Review by jw3126 — COMMENTED** [review]
While I think the formatting does many micro improvements like adding a `;` before keyword arguments it destroys some more large scale structures like alignment of `=` signs in consecutive lines or some longish but ok one-liners. Overall I think it is easier for a small project like this to maintain code format manually than using the formatter.

**Pull request closed without merging**